jme3-core/src/main/java/com/jme3/anim/AnimComposer.java

```diff
@@ -456,19 +421,59 @@ public void setGlobalSpeed(float globalSpeed) {
         this.globalSpeed = globalSpeed;
     }
 
+    /**
+     * Provides access to the named layer.
+     *
+     * @param layerName the name of the layer to access
+     * @return the pre-existing instance
+     */
+    public AnimLayer getLayer(String layerName) {
+        AnimLayer result = layers.get(layerName);
+        if (result == null) {
+            throw new IllegalArgumentException("Unknown layer " + layerName);
+        }
+        return result;
+    }
+
     /**
      * Access the manager of the named layer.
      *
      * @param layerName the name of the layer to access
      * @return the current manager (typically an AnimEvent) or null for none
      */
     public Object getLayerManager(String layerName) {
-        Layer layer = layers.get(layerName);
-        if (layer == null) {
-            throw new IllegalArgumentException("Unknown layer " + layerName);
+        AnimLayer layer = getLayer(layerName);
+        Object result = layer.getManager();
+
+        return result;
+    }
+
+    /**
+     * Find the name of the specified layer.
+     *
+     * @param layer the layer to find
+     * @return the layer's name, or null if not found
+     */
+    public String getLayerName(AnimLayer layer) {
```

Comment:
the behavior of this method is a bit unusual, why don't we add the 'name' attribute to the AnimLayer class instead of managing it with a map?
We could do an optimization: add the 'name' attribute to the AnimLayer class + public getter method, and replace the map with a list in the AnimComposer class. These are all internal changes that do not impact the external use of the classes.

Reply:
I considered adding the name to `AnimLayer`. I'm willing to do that change. However, I don't think a `List` is a good idea because it's inefficient in the common case, which is looking up the layer for a given name. There's no limit on the number of layers in an `AnimControl`.

Reply:
ok for the map, but we add the name attribute to the AnimLayer class so this method is no longer needed

Reply:
yes